tests/rbd/test_scale_group_snap_clone.py

```diff
@@ -0,0 +1,241 @@
+    group_info,
+    group_snap_info,
+)
+from ceph.rbd.workflows.krbd_io_handler import krbd_io_handler
+from ceph.utils import get_node_by_id
+from cli.rbd.rbd import Rbd
+from utility.log import Log
+
+log = Log(__name__)
+
+
+def test_scale_group_snap_clone(rbd_obj, client, **kw):
+    """
+    Tests the group creation followed by cloning image from group snapshot at scale
+    Args:
+        client: rbd client obj
+        **kw: test data
+
+    Test Steps:
+    1. Deploy Ceph on version 8.0 or greater.
+    2. Create an RBD pool named pool1.
+    3. Create more than 100 RBD images in the pool using a loop.
+    4. Create an RBD group named group1 in pool1 and
+    5. Add all 100+ images to the group using a loop.
+    6. Take a snapshot of the group using the rbd group snap create command.
+    7. Verify the group snapshot exists using the rbd group snap ls command.
+    8. Clone the group snapshot for all 100+ images using the rbd clone --snap-id option
+    9. Verify the cloned images exist in the pool and check their parent image using rbd info in a loop.
+    10. Map the cloned images as new block disks and run fio using rbd map in a loop.
+    """
+    kw["client"] = client
+    rbd = rbd_obj.get("rbd")
+    rbd1 = Rbd(kw["client"])
+    fio = kw.get("config", {}).get("fio", {})
+
+    for pool_type in rbd_obj.get("pool_types"):
+        rbd_config = kw.get("config", {}).get(pool_type, {})
+        multi_pool_config = getdict(rbd_config)
+        for pool, pool_config in multi_pool_config.items():
+            log.info(f"All pool details : {pool} and {pool_config}")
+            if "data_pool" in pool_config.keys():
+                _ = pool_config.pop("data_pool")
+
+            # Create a group per pool
+            log.info(f"Create a group {pool}")
+            group = kw.get("config", {}).get("group", "image_group_default")
+            group_create_kw = {"client": client, "pool": pool, "group": group}
+            rc = create_group_and_verify(**group_create_kw)
+            if rc:
+                log.error(f"group {group} creation failed, test case fail")
+                return 1
+            else:
+                log.info("STAGE: group creation succeeded")
+
+            for image, image_config in pool_config.items():
+                log.info(f"All image details : {image} and {image_config}")
+
+                # Add all 100+ images to the group using a loop.
+                log.info(f"Adding image: {image} in group {group}")
+                add_image_group_kw = {
+                    "client": client,
+                    "pool": pool,
+                    "group": group,
+                    "image": image,
+                }
+                rc = add_image_to_group_and_verify(**add_image_group_kw)
+                if rc:
+                    log.error(f"Image {image} add failed, test case fail")
+                    return 1
+                else:
+                    log.info(f"STAGE: image {image} add succeeded")
```

Comment:
can it be `STAGE: image {image} added successfully to group`

Reply:
Done